src/NHibernate/Linq/Assignments.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using NHibernate.Linq.Visitors;
+
+namespace NHibernate.Linq
+{
+	/// <summary>
+	/// Class to hold assignments used in updates and inserts.
+	/// </summary>
+	/// <typeparam name="TSource">The type of the entity source of the insert or to update.</typeparam>
+	/// <typeparam name="TTarget">The type of the entity to insert or to update.</typeparam>
+	public class Assignments<TSource, TTarget>
+	{
+		private readonly Dictionary<string, Expression> _assignments = new Dictionary<string, Expression>();
+
+		internal IReadOnlyDictionary<string, Expression> List => _assignments;
+
+		/// <summary>
+		/// Sets the specified property.
+		/// </summary>
+		/// <typeparam name="TProp">The type of the property.</typeparam>
+		/// <param name="property">The property.</param>
+		/// <param name="expression">The expression that should be assigned to the property.</param>
+		/// <returns>The current assignments list.</returns>
+		public Assignments<TSource, TTarget> Set<TProp>(Expression<Func<TTarget, TProp>> property, Expression<Func<TSource, TProp>> expression)
```

Comment:
Am I wrong or `SqlFunctions` are currently completely unsupported? They do not fit well NHibernate anyway, as they seems specifically meant for SQL Server. NHibernate does not promote using database specific feature usually, it rather supplies database agnostic equivalent.

At this subject, I have put in my todo list since long: NH- : a static class like `HqlFunctions` supplying a way to call with Linq built-in HQL function supported by dialects.
At least having a `HqlFunctions.Call(string name, params object[] args)` would be handy. But there are many other things in my list.

Overall, is there currently a supported case where the "no args" lambda would be useful? If none, I would rather not add a "not yet useful method".